### ISSUE TITLE
Option to turn on/off hardware acceleration 

### ISSUE THREAD

**Issue by austinhuang0131** [vector-im/element-web#13648]
**Is your suggestion related to a problem? Please describe.**
Mac tends to freeze a lot when hardware acceleration is frequently used. A known example is [patch 2020-002](https://mrmacintosh.com/2020-002-update-causes-some-macs-to-freeze-when-using-hw-accelerated-video/). Also this would help low-end devices running the app.

**Describe the solution you'd like**
A switch to enable/disable hardware acceleration for graphics.

**Describe alternatives you've considered**
The switch certainly exists for Discord (which is also electron)...

**Additional context**

edit: typo

**Comment by austinhuang0131** [comment]
Also, my computer (MacBook Air) froze with that specific fault (in linked article) when launching the YouTube widget, so that sorta makes the issue somewhere between an issue and a suggestion...

**Comment by turt2live** [comment]
that would probably be a different issue, though likely to be closed as a duplicate of this due to lack of the feature in the first place.

**Comment by novocaine** [comment]
> Seems this is still relevant, since hardware acceleration is causing #21405 It would be great to have an option for this. All it needs to do is call `app.disableHardwareAcceleration()` at startup

You (or the OP) could verify whether your suggestion indeed resolves https://github.com/vector-im/element-web/issues/21405 by running `app.disableHardwareAcceleration()` in the developer console (ctrl/cmd+alt+I). 

Electron's documentation [doesn't draw that conclusion](https://github.com/electron/electron/blob/main/docs/faq.md#the-font-looks-blurry-what-is-this-and-what-can-i-do), and 
 in the past, we've found it [hasn't been that simple](https://github.com/vector-im/element-web/issues/15594)

**Comment by MRAAGH** [comment]
https://www.electronjs.org/docs/latest/api/app#appdisablehardwareacceleration states that this can only be called before the app is ready; so I don't think we can do this while it is already running. Additionally, I don't seem to be able to find `app`, it either exists under a different name or is not globally defined
![Screenshot_20220316-181028_r8KQb](https://user-images.githubusercontent.com/33424247/158647938-cbfafcff-f1f8-48d3-885b-af7390ebd8f5.png)

**Comment by novocaine** [comment]
> https://www.electronjs.org/docs/latest/api/app#appdisablehardwareacceleration states that this can only be called before the app is ready; so I don't think we can do this while it is already running. Additionally, I don't seem to be able to find `app`, it either exists under a different name or is not globally defined ![Screenshot_20220316-181028_r8KQb](https://user-images.githubusercontent.com/33424247/158647938-cbfafcff-f1f8-48d3-885b-af7390ebd8f5.png)

ah right. well, its not too complicated to to a local build and hack the line in to test, instructions are over at https://github.com/vector-im/element-desktop/

**Comment by SapphyThyHybrid** [comment]
> > Seems this is still relevant, since hardware acceleration is causing #21405 It would be great to have an option for this. All it needs to do is call `app.disableHardwareAcceleration()` at startup
> 
> You (or the OP) could verify whether your suggestion indeed resolves #21405 by running `app.disableHardwareAcceleration()` in the developer console (ctrl/cmd+alt+I). If you can show that, it would help motivate this enhancement request.
> 
> Note that Electron's documentation [doesn't draw that conclusion](https://github.com/electron/electron/blob/main/docs/faq.md#the-font-looks-blurry-what-is-this-and-what-can-i-do), and in the past, we've found it [hasn't been that simple](https://github.com/vector-im/element-web/issues/15594).

I'll do my best then

**Comment by t3chguy** [comment]
> Additionally, I don't seem to be able to find app, it either exists under a different name or is not globally defined

Its only available from the Main process, not the Renderer process.

![image](https://user-images.githubusercontent.com/2403652/159236731-65c5acda-2954-4d06-ac95-c5de6b5b11ee.png)

**Comment by MRAAGH** [comment]
I am not personally experiencing this issue, but for anyone who is, here is how you can test if disabling hardware acceleration is the solution:
1. install yarn (for example from https://github.com/yarnpkg/yarn/releases)
2. go to https://github.com/vector-im/element-desktop/ click Code and then Download ZIP. Extract the zip
3. open a terminal in the extracted folder and run these commands:
```
yarn install
yarn run fetch --noverify --cfgdir ""
yarn add electron
yarn start
```
Your freshly compiled Element should start. Please check that your issue still happens in this version. Then quit Element.

4. open the folder `src` and edit `electron-main.ts`. Find the line that says `app.on('ready', async () => {` (should be line number 846). Before this line, add the following line:
```
app.disableHardwareAcceleration();
```
Save and close.

5. Run the command `yarn start` again. Element starts again. Check that the issue does not happen any more.

**Comment by SapphyThyHybrid** [comment]
> > > Seems this is still relevant, since hardware acceleration is causing #21405 It would be great to have an option for this. All it needs to do is call `app.disableHardwareAcceleration()` at startup
> > 
> > 
> > You (or the OP) could verify whether your suggestion indeed resolves #21405 by running `app.disableHardwareAcceleration()` in the developer console (ctrl/cmd+alt+I). If you can show that, it would help motivate this enhancement request.
> > Note that Electron's documentation [doesn't draw that conclusion](https://github.com/electron/electron/blob/main/docs/faq.md#the-font-looks-blurry-what-is-this-and-what-can-i-do), and in the past, we've found it [hasn't been that simple](https://github.com/vector-im/element-web/issues/15594).
> 
> I'll do my best then

Continuing with this and with the help of friends, eventually we go there, thanks to this person and these steps (as well as other cause windows wants to be windows

> I am not personally experiencing this issue, but for anyone who is, here is how you can test if disabling hardware acceleration is the solution:
> 
> 1. install yarn (for example from https://github.com/yarnpkg/yarn/releases)
> 2. go to https://github.com/vector-im/element-desktop/ click Code and then Download ZIP. Extract the zip
> 3. open a terminal in the extracted folder and run these commands:
> 
> ```
> yarn install
> yarn run fetch --noverify --cfgdir ""
> yarn add electron
> yarn start
> ```
> 
> Your freshly compiled Element should start. Please check that your issue still happens in this version. Then quit Element.
> 
> 4. open the folder `src` and edit `electron-main.ts`. Find the line that says `app.on('ready', async () => {` (should be line number 846). Before this line, add the following line:
> 
> ```
> app.disableHardwareAcceleration();
> ```
> 
> Save and close.
> 
> 5. Run the command `yarn start` again. Element starts again. Check that the issue does not happen any more.

disabling hardware acceleration has fixed the issue for me

Before disabling:
![image](https://user-images.githubusercontent.com/101528021/159808709-db236272-ee3b-40ce-a76a-c7d230e6022c.png)

and after disabling:
![image](https://user-images.githubusercontent.com/101528021/159808742-9642108c-f96b-435e-9919-a89fb49e6624.png)
I waited a couple of mins after disabling to see if it comes back (usually takes about 10 seconds for it to become blurry for me) and it stayed readable. I hope this is enough proof that disabling hardware acceleration fixes the issue for me, and sorry it took a while. Life happened.

**Comment by MRAAGH** [comment]
tl;dr: yes, `app.disableHardwareAcceleration();` fixed everything, and there should probably be a setting for this.

**Comment by MRAAGH** [comment]
:tada:
